src/files-and-videos/videos/transcript-settings/OrderTranscriptForm.jsx

```diff
@@ -0,0 +1,165 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Button, SelectableBox, Stack } from '@edx/paragon';
+import { ErrorAlert } from '@edx/frontend-lib-content-components';
+import Cielo24Form from './Cielo24Form';
+import ThreePlayMediaForm from './ThreePlayMediaForm';
+import { RequestStatus } from '../../../data/constants';
+import messages from './messages';
+import { checkCredentials, validateForm } from '../data/utils';
+
+const OrderTranscriptForm = ({
+  setTranscriptType,
+  activeTranscriptPreferences,
+  transcriptType,
+  transcriptCredentials,
+  closeTranscriptSettings,
+  handleOrderTranscripts,
+  transcriptionPlans,
+  errorMessages,
+  transcriptStatus,
+  // injected
+  intl,
+}) => {
+  const [data, setData] = useState(activeTranscriptPreferences || { videoSourceLanguage: '' });
+
+  let [cieloHasCredentials, threePlayHasCredentials] = checkCredentials(transcriptCredentials);
+  useEffect(() => {
+    [cieloHasCredentials, threePlayHasCredentials] = checkCredentials(transcriptCredentials);
+  }, [transcriptCredentials]);
+
+  let isFormValid = validateForm(cieloHasCredentials, threePlayHasCredentials, transcriptType, data);
+  useEffect(() => {
+    isFormValid = validateForm(cieloHasCredentials, threePlayHasCredentials, transcriptType, data);
+  }, [data]);
+
+  const handleDiscard = () => {
+    setTranscriptType(activeTranscriptPreferences);
+    closeTranscriptSettings();
+  };
+
+  const handleUpdate = () => handleOrderTranscripts(data, transcriptType);
+
+  let form;
+  switch (transcriptType) {
+  case 'Cielo24':
+    form = (
+      <Cielo24Form
+        {...{
+          hasTranscriptCredentials: cieloHasCredentials,
+          data,
+          setData,
+          transcriptionPlan: transcriptionPlans.Cielo24,
+        }}
+      />
+    );
+    break;
+  case '3PlayMedia':
+    form = (
+      <ThreePlayMediaForm
+        {...{
+          hasTranscriptCredentials: threePlayHasCredentials,
+          data,
+          setData,
+          transcriptionPlan: transcriptionPlans['3PlayMedia'],
+        }}
+      />
+    );
+    break;
+  default:
+    break;
+  }
+  return (
+    <>
+      <ErrorAlert
+        hideHeading={false}
+        isError={transcriptStatus === RequestStatus.FAILED}
+      >
+        <ul className="p-0">
+          {errorMessages.transcript.map(message => (
+            <li key={`add-error-${message}`} style={{ listStyle: 'none' }}>
+              {intl.formatMessage(messages.errorAlertMessage, { message })}
+            </li>
+          ))}
+        </ul>
+      </ErrorAlert>
```

Comment:
I'm not sure why github only shows line 81 as not covered, but I don't see any test that covers this error alert for `RequestStatus.FAILED`. Then probably you want to make sure that a test also contains an `errorMessages.transcript` array here. And out of curiosity, can the `intl.formatMessage` method handle the case where `messages.errorAlertMessage` is undefined? If not, this line also needs to be unit tested (you can extract it to a function and just add one jest test for it).

Reply:
If it's just about testing with `RequestStatus.FAILED`, it's probably sufficient to cover this via a `TranscriptSettings.test.jsx` test.

Reply:
Regarding `messages.errorAlertMessage`, that is always defined because it comes from `messages.js`. If `message` is undefined, the intl will return a formatted message of "undefined".